airbyte-bootloader/src/main/java/io/airbyte/bootloader/Bootloader.java

```diff
@@ -0,0 +1,291 @@
+                    final AirbyteVersion currentAirbyteVersion,
+                    final Optional<DefinitionsProvider> definitionsProvider,
+                    final FeatureFlags featureFlags,
+                    @Named("jobsDatabaseInitializer") final DatabaseInitializer jobsDatabaseInitializer,
+                    @Named("jobsDatabaseMigrator") final DatabaseMigrator jobsDatabaseMigrator,
+                    final JobPersistence jobPersistence,
+                    final ProtocolVersionChecker protocolVersionChecker,
+                    @Value("${airbyte.run-migration-on-startup}") final boolean runMigrationOnStartup,
+                    final SecretMigrator secretMigrator) {
+    this(autoUpgradeConnectors, configRepository, configsDatabaseInitializer, configsDatabaseMigrator, currentAirbyteVersion,
+        definitionsProvider, featureFlags, jobsDatabaseInitializer, jobsDatabaseMigrator, jobPersistence, protocolVersionChecker,
+        runMigrationOnStartup, secretMigrator, null);
+  }
+
+  @VisibleForTesting
+  Bootloader(
+             @Value("${airbyte.protocol.auto-upgrade-connectors}") final boolean autoUpgradeConnectors,
+             final ConfigRepository configRepository,
+             @Named("configsDatabaseInitializer") final DatabaseInitializer configsDatabaseInitializer,
+             @Named("configsDatabaseMigrator") final DatabaseMigrator configsDatabaseMigrator,
+             final AirbyteVersion currentAirbyteVersion,
+             final Optional<DefinitionsProvider> definitionsProvider,
+             final FeatureFlags featureFlags,
+             @Named("jobsDatabaseInitializer") final DatabaseInitializer jobsDatabaseInitializer,
+             @Named("jobsDatabaseMigrator") final DatabaseMigrator jobsDatabaseMigrator,
+             final JobPersistence jobPersistence,
+             final ProtocolVersionChecker protocolVersionChecker,
+             @Value("${airbyte.run-migration-on-startup}") final boolean runMigrationOnStartup,
+             final SecretMigrator secretMigrator,
+             final Runnable postLoadExecution) {
+    this.autoUpgradeConnectors = autoUpgradeConnectors;
+    this.configRepository = configRepository;
+    this.configsDatabaseInitializer = configsDatabaseInitializer;
+    this.configsDatabaseMigrator = configsDatabaseMigrator;
+    this.currentAirbyteVersion = currentAirbyteVersion;
+    this.definitionsProvider = definitionsProvider;
+    this.featureFlags = featureFlags;
+    this.jobsDatabaseInitializer = jobsDatabaseInitializer;
+    this.jobsDatabaseMigrator = jobsDatabaseMigrator;
+    this.jobPersistence = jobPersistence;
+    this.protocolVersionChecker = protocolVersionChecker;
+    this.runMigrationOnStartup = runMigrationOnStartup;
+    this.secretMigrator = secretMigrator;
+    this.postLoadExecution = postLoadExecution;
+  }
+
+  /**
+   * Ensures additional initialization is performed after all dependency injection has completed.
+   */
+  @PostConstruct
```

Comment:
I don't think we can have this as a postConstruct.
We have a few validation in the `.load` that may decide to abort the process. I understand that `PostConstruct` would run before the `.load` which means that we would always update connectors even if we decide to abort the bootloader.

Reply:
We are only setting the variable to the lambda/runnable, not executing it in the post construct.  That being said, I think it makes sense to create a class that represents the runnable so we can treat it like a singleton and replace it more easily in cloud.

Reply:
Indeed, I read it as apply the function rather than set.